libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -214,15 +214,15 @@ AP_GPS::detect_instance(uint8_t instance)
     if (_type[instance] == GPS_TYPE_PX4) {
         // check for explicitely chosen PX4 GPS beforehand
         // it is not possible to autodetect it, nor does it require a real UART
-        _broadcast_gps_type("PX4", instance, -1); // baud rate isn't valid
+        dstate->last_baud = -1; // baud rate isn't valid
         new_gps = new AP_GPS_PX4(*this, state[instance], _port[instance]);
         goto found_gps;
     }
 #endif
 
 #if CONFIG_HAL_BOARD == HAL_BOARD_QURT
     if (_type[instance] == GPS_TYPE_QURT) {
-        _broadcast_gps_type("QURTGPS", instance, -1); // baud rate isn't valid
+        dstate->last_baud = -1; // baud rate isn't valid
```

Comment:
This is unneeded, _broadcast_gps_type() didn't use to set it, and it doesn't matter for this GPS type, and can be entirely ignored.